---
title: "Report"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning=FALSE, message=FALSE)
library(knitr)
library(ggplot2)
library(tidyr)
library(devtools)
library(pander)
library(rmarkdown)
#getwd()
```

## Result of simulation

This is html report for siumaltion reads to calculate FDR, TP, FP, TN, FN.

```{r report1 result by each of genome, echo=F}
report1=read.csv('test.csv',sep='\t',header=F)
colnames(report1)=c("gene","read_count","length","coverage")
#report1$acc_no = 

t=separate(data = report1, col = gene, into = c("left", "right"), sep = "_")
t=as.data.frame(table(t$left))
colnames(t) = c("Gene","No_of_detected_prot")
t$tot_prot = c(11,11)
t$pct_detect=round(t$No_of_detected_prot/t$tot_prot*100,2)
#barplot()
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
pander(t)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
