---
title: "Report"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning=FALSE, message=FALSE)
library(knitr)
library(ggplot2)
library(tidyr)
library(devtools)
library(pander)
library(rmarkdown)
#getwd()
```

## Result of simulation

### Simulation for each of reference genome

```{r report1 result by each of genome, echo=F}

report1=read.csv('summarise.csv',sep='\t',header=F)

colnames(report1)=c("gene","read_count","length","coverage")
#report1$acc_no = 

prot_dict = read.csv('access_vs_prot.csv',sep=',',header=F)
colnames(prot_dict)=c("acc_id","no_prot")
t=separate(data = report1, col = gene, into = c("Acc_no", "Protein"), sep = "_")
t=as.data.frame(table(t$Acc_no))

colnames(t) = c("acc_id","No_of_detected_prot")
x=merge(t,prot_dict,by="acc_id",all.x=T)
x$pct_detect=round(x$No_of_detected_prot/x$no_prot*100,2)
#barplot()
```

Table below is resulted from simulating each reference genome, generating 10000 reads, each read is 100bp with zero mutation rate and 0.5% error rate when aligning one by one reference genome separately to the graph:

```{r pressure, echo=FALSE}
pander(x)
```
### Plots


```{r plot for detection, echo = F}
plot(x$pct_detect,ylim = c(60,100), ylab="% detected protein",pch=20,col="blue")
lines(x$pct_detect,col="grey")
```

### Simulation for mixture of 3 reference genomes


### Simulation for mixture of contaminations


### Simulation for mixture of reference and contaminations







Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
